oci/archive/oci_transport.go

```diff
@@ -0,0 +1,179 @@
+package archive
+
+import (
+	"io/ioutil"
+	"regexp"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/containers/image/directory/explicitfilepath"
+	"github.com/containers/image/docker/reference"
+	"github.com/containers/image/image"
+	ocilayout "github.com/containers/image/oci/layout"
+	"github.com/containers/image/transports"
+	"github.com/containers/image/types"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	transports.Register(Transport)
+}
+
+// Transport is an ImageTransport for OCI archive
+// it creates an oci-archive tar file by calling into the OCI transport
+// tarring the directory created by oci and deleting the directory
+var Transport = ociArchiveTransport{}
+
+type ociArchiveTransport struct{}
+
+// ociArchiveReference is an ImageReference for OCI Archive paths
+type ociArchiveReference struct {
+	resolvedFile  string
+	tag           string
+	tempDirectory string
+	ociRef        types.ImageReference
+}
+
+func (t ociArchiveTransport) Name() string {
+	return "oci-archive"
+}
+
+// ParseReference converts a string, which should not start with the ImageTransport.Name prefix
+// into an ImageReference.
+func (t ociArchiveTransport) ParseReference(reference string) (types.ImageReference, error) {
+	return ParseReference(reference)
+}
+
+// ValidatePolicyConfigurationScope checks that scope is a valid name for a signature.PolicyTransportScopes keys
+func (t ociArchiveTransport) ValidatePolicyConfigurationScope(scope string) error {
+	return ocilayout.Transport.ValidatePolicyConfigurationScope(scope)
```

Comment:
Still mixing the semantics… either copy&paste `layout.Transport.ValidatePolicyConfigurationScope`, or extract into a shared helper which is explicitly agnostic of dir/path (i.e. rename existing `layout.Transport.ValidatePolicyConfigurationScope` to something like `oci.ValidatePathPolicyConfigurationScope`, and then call it from both `{archive,layout}.Transport.ValidatePOlicyConfigurationScope`